--- soci-3.0.0/src/backends/mysql/common.h.org	2008-07-10 19:37:12.388879700 +0200
+++ soci-3.0.0/src/backends/mysql/common.h	2008-07-10 19:37:22.405876279 +0200
@@ -9,6 +9,7 @@
 #define SOCI_MYSQL_COMMON_H_INCLUDED
 
 #include <sstream>
+#include <ctime>
 #include "soci-mysql.h"
 
 namespace soci
--- soci-3.0.0/src/backends/mysql/session.cpp.org	2008-07-10 19:44:41.317881830 +0200
+++ soci-3.0.0/src/backends/mysql/session.cpp	2008-07-10 19:46:09.761878328 +0200
@@ -10,6 +10,8 @@
 #include "soci-mysql.h"
 #include <cctype>
 #include <cerrno>
+#include <climits>
+#include <cstdlib>
 #include <ciso646>
 
 
--- soci-3.0.0/src/backends/mysql/standard-into-type.cpp.org	2008-07-10 19:48:28.296879993 +0200
+++ soci-3.0.0/src/backends/mysql/standard-into-type.cpp	2008-07-10 19:49:06.593880673 +0200
@@ -12,6 +12,7 @@
 #include "common.h"
 #include <ciso646>
 #include <cassert>
+#include <cstring>
 
 #ifdef _MSC_VER
 #pragma warning(disable:4355)
--- soci-3.0.0/src/core/soci-simple.cpp~	2008-07-01 09:42:25.000000000 +0200
+++ soci-3.0.0/src/core/soci-simple.cpp	2011-05-17 09:15:12.366747904 +0200
@@ -7,6 +7,8 @@
 
 #define SOCI_SOURCE
 
+#include <cstdio>
+
 #include "soci-simple.h"
 #include "soci.h"
 
--- soci-3.0.0/src/backends/mysql/standard-use-type.cpp~	2008-05-15 15:58:38.000000000 +0200
+++ soci-3.0.0/src/backends/mysql/standard-use-type.cpp	2011-05-17 09:17:43.470359903 +0200
@@ -13,6 +13,7 @@
 #include <ciso646>
 #include <limits>
 #include <cstring>
+#include <cstdio>
 
 #ifdef _MSC_VER
 #pragma warning(disable:4355)
--- soci-3.0.0/src/backends/mysql/vector-use-type.cpp~	2008-05-15 15:58:38.000000000 +0200
+++ soci-3.0.0/src/backends/mysql/vector-use-type.cpp	2011-05-17 09:19:24.358330530 +0200
@@ -13,6 +13,7 @@
 #include <ciso646>
 #include <limits>
 #include <cstring>
+#include <cstdio>
 
 #ifdef _MSC_VER
 #pragma warning(disable:4355)
